src/polycubed/src/controller.cpp

```diff
@@ -418,14 +436,30 @@ void Controller::unregister_cb(int id) {
 
 // caller must guarantee that module_index and port_id are valid
 void Controller::send_packet_to_cube(uint16_t module_index, uint16_t port_id,
-                                     const std::vector<uint8_t> &packet) {
+                                     const std::vector<uint8_t> &packet,
+                                     service::Sense sense, bool is_tr_cube) {
   ctrl_rx_md_index_++;
   ctrl_rx_md_index_ %= MD_MAP_SIZE;
 
-  metadata md_temp = {module_index, port_id};
+  metadata md_temp = {module_index, port_id, MD_PKT_FROM_CONTROLLER};
+  if (sense == service::Sense::EGRESS) {
+      md_temp.flags |= MD_EGRESS_CONTEXT;
+  }
   metadata_table_->update_value(ctrl_rx_md_index_, md_temp);
 
-  iface_->send(const_cast<std::vector<uint8_t> &>(packet));
+  if (!is_tr_cube || sense == service::Sense::EGRESS) {
+      iface_->send(const_cast<std::vector<uint8_t> &>(packet));
+  } else {
+      /* for transparent cube ingress packet needs modifying the
```

Comment:
Should the MAC also be changed when the transparent cube is attached to another cube's port?

Reply:
I didn't test the regular cube but I think the MAC shouldn't impact the regular cube processing because there is no chance the packet back to kernel stack in regular cube case.

Reply:
Actually I am not worried about the stack case but more about breaking standard cubes. What about if the standard cube uses the MAC of the packet to do any calculation?.

Reply:
Oh, I understood your point and modified the code limiting the scope to transparent && ingress context && attached to a interface.